### PR TITLE
Move data internalisation to forward models

### PR DESCRIPTION
**Pull request by pinkwah**
Resolves: #6035

### CODE REVIEW

File: src/ert/storage/local_storage.py [hunk 1]
```diff
@@ -194,21 +195,26 @@ def __init__(
...
+                    " You may already be running ERT,"
+                    " or another user is using the same ENSPATH."
+                ) from e
 
         super().__init__(path)
 
         self._save_index()
 
     def close(self) -> None:
+        if self._ignore_mutex:
```
Comment (berland) [inline src/ert/storage/local_storage.py:215]: This function is doing "nothing" ??

Comment (pinkwah) [reply src/ert/storage/local_storage.py:215]: If `self._ignore_mutex` is set, yes. Otherwise it'll save data to disk.

Comment (pinkwah) [reply src/ert/storage/local_storage.py]: This change has been since removed.

File: src/ert/job_queue/job_queue_node.py [hunk 2]
```diff
@@ -275,12 +249,35 @@ def _handle_end_status(
     ) -> None:
         with self._mutex:
             if end_status == JobStatus.DONE:
-                with pool_sema:
-                    logger.info(
-                        f"Realization: {self.run_arg.iens} complete, "
-                        "starting to load results"
+                success = True
+                if driver.name == "LOCAL":
```
Comment (xjules) [inline src/ert/job_queue/job_queue_node.py:253]: Can we use QueueSystem enum instead?

**Comment by berland** [comment]
When running on a local queue, the entry still ends up in jobs.json (but it might not do anything, I observe the GUI reports its duration to 0 seconds?). We probably don't want it listed there when it is not doing anything.

**Comment by berland** [comment]
I have tried this PR on "bigpoly". On plain bigpoly (100 realz,  ~200 FORWARD_MODELs, big summary file) it is no faster than the main branch (ensemble_experiment finishes in 3 minutes).

If a `FIELD` keyword is added to bigpoly file (the same FIELD as is generated by the bigpoly yaml github action workflow), then the ensemble_experiment still takes 3 minutes, but our `main` branch uses around 30 minutes 👏🏻  (I stopped it at 78 out of 100 realizations at 22 minutes, didn't bother to wait for it)

**Comment by berland** [comment]
When running on the poly-case with `OBS_CONFIG` and `GEN_DATA` commented out, there is nothing to internalize, but this PR still adds to `jobs.json`. It probably shouldn't.

**Comment by eivindjahren** [comment]
I think this is a nice experiment that has shown us how this behaves for files of various sizes.

It's probably worth trying other experiments that may give better performance gains. One we discussed in the office was to always save the `.nc` file that ends up in storage to runpath first and let `_ert_job_runner` inflate it to e.g. `grdecl`, `irap`, etc. That way you would not have to import anything from ert to inflate as `_ert_job_runner` just needs to be able to read the information required to inflate from the `.nc` file. That the contract of what goes into the `.nc` file is kept between `_ert_job_runner` and `ert` is trivial to test.

**Comment by oyvindeide** [comment]
Closing this as it looks stale and has a bunch of conflicting files, feel free to reopen if you pick it up again.

**Comment by pinkwah** [comment]
Closing. Will make another attempt from first principles.

**Pull request closed without merging**